libraries/SITL/SIM_JEDEC.cpp

```diff
@@ -0,0 +1,193 @@
+#include "SIM_JEDEC.h"
+
+#include <errno.h>
+#include <unistd.h>
+
+#include <AP_HAL_SITL/AP_HAL_SITL.h>
+
+using namespace SITL;
+
+extern HAL_SITL& hal;
+
+void JEDEC::open_storage_fd()
+{
+    if (storage_fd != -1) {
+        AP_HAL::panic("Should not have been called");
+    }
+    const char *filepath = filename();
+    uint32_t flags = O_RDWR|O_CREAT;
+    if (hal.get_wipe_storage()) {
+        unlink(filepath);
+    }
+    storage_fd = open(filepath, flags, 0644);
+    if (storage_fd == -1) {
+        AP_HAL::panic("open(%s): %s", filepath, strerror(errno));
+    }
+    if (ftruncate(storage_fd, storage_size()) != 0) {
+        AP_HAL::panic("truncate(%s): %s", filepath, strerror(errno));
+    }
+}
+
+void JEDEC::sector4k_erase (uint32_t addr)
+{
+    for (uint8_t i=0; i<get_page_per_sector(); i++) {
+        page_erase(addr + i*get_page_size());
```

Comment:
`page_erase` could take a page number, and `sector4k` could take a sector number - and check their alignment.

Same  with `block64k_erase`

Doesn't need to be done as part of this PR.